### DISCUSSION TITLE
Question: Can instrospection change table names to something else? like camelCased?

### DISCUSSION THREAD

**Discussion by hcharley** [prisma/prisma#1925] [Q&A] [answered]
Hi, 

I'm wondering specifically about tableNames -> modelNames. I'd prefer pascal case or camel case to snake case (what i write table names in).

I'd also extend this to column names as well.

**Comment by pantharshit00** [comment, marked as answer]
Yes, you can change the introspected table names using the `@@map` directive.

Example:
```prisma
model User {
  id   String @id @default(cuid())
  name String
  @@map("user")
}
```

**Comment by nikolasburk** [comment]
@charlex we have a [dedicated docs page](https://prisma2.netlify.com/reference/tools-and-interfaces/prisma-client/configuring-the-prisma-client-api) for that in the new docs (which will be published soon)

**Comment by hcharley** [comment]
I see, thank you! 

I'm wondering how hard it would be to transform the model names to pascal case _automatically_ as part of the introspection.

Part of my goal is to simply have the Prisma2 schema be an always-generated output of the database.

I suppose I could write a script that would just help me do what the docs suggest to do manually.

I almost wonder if introspection should do this by default to help users conform to Prisma's naming conventions.

**Reply by hcharley** [reply]
Just going to dump some regex here that might help write that script:

```regex
model ([a-z_]+) {
```

```ts
const result = `
model ${pascalCase(name)} {
  @@map(${name})
}
`
```

**Reply by nikolasburk** [reply]
This will be possible in the future! 🙂  We are already working on a [spec](https://github.com/prisma/specs/tree/introspection-configuration/cli#introspection-configuration) for that 🙌

**Reply by nikolasburk** [reply]
(Note that the casing isn't yet part of the spec, but this will be enabled by the introspection configuration that's described there).

**Reply by hcharley** [reply]
Nice, then I will probably pause on my own implementation. Thanks for pointing me in the right direction on that!

**Reply by nikolasburk** [reply]
Actually would be great if you could [create a feature request](https://github.com/prisma/prisma2/issues/new?assignees=&labels=&template=feature_request.md&title=) with your exact use case and requirements (just so we can make sure to consider this when we design that feature). 🙂

**Reply by hcharley** [reply]
Done: https://github.com/prisma/prisma2/issues/1934

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
